spec/models/user_spec.rb

```diff
@@ -367,4 +366,21 @@
       it { expect(subject.admin?).to be_truthy }
     end
   end
+
+  describe '#full_name' do
+    subject { create(:user, first_name: 'first', last_name: 'last') }
+
+    it 'should compose #first_name' do
+      expect(subject.first_name).to eq('first')
+    end
+    it 'should compose #last_name' do
+      expect(subject.last_name).to eq('last')
+    end
+    it 'should return "first_name last_name"' do
+      expect(subject.full_name).to eq('first last')
+    end
+    it 'should belong to a valid User' do
+      expect(subject.valid?).to be true
+    end
```

Comment:
@matthew-campbell - I know this may be a bit of a stretch for this story, but reviewing your tests reminded me that we are not testing validations, and perhaps we should add a section for those.

That is, the model has:
```
validates_presence_of :first_name, :last_name
```
Yet we don't test for that.  We can use shoulda matchers for that - see `validate_presence_of` here: http://matchers.shoulda.io/docs/v3.1.1/. (It appears that we _are_ doing that for other models.)

I suggest adding a section `describe 'Validations'` to the file and adding these tests.

Reply:
I would be happy to update to shoulda if you think that will be better than what I have suggested. I only took a brief look at it and saw this (lack of maintainers) note from its maintainer:

https://github.com/thoughtbot/shoulda-matchers#note-from-the-maintainer